src/npcmove.cpp

```diff
@@ -1716,6 +1792,60 @@ static tripoint nearest_passable( const tripoint &p, const tripoint &closest_to
     return tripoint_min;
 }
 
+
+void npc::move_away_from( const std::vector<tripoint> &points, int safe_distance, bool no_bashing )
+{
+    if( points.empty() ) {
+        return;
+    }
+
+    const auto candidate_points = closest_tripoints_first( safe_distance, pos() );
+    int least_danger = std::numeric_limits<int>::max();
+    int cheapest_path = std::numeric_limits<int>::max();
+
+    tripoint best_hideout = pos();
+
+    for( const auto &elem : candidate_points ) {
```

Comment:
This part could be expensive since there is a lot of overdraw.
Test it with a semi-winding path and 3+ NPCs. If it doesn't chug, then it's fine.
If it does, you could skip the actual pathfinding and just implement "inverse Dijkstra" here. That is, a flood fill with the checks you have here. Then call `update_path` once, at the end.

Reply:
> If it doesn't chug, then it's fine.

It didn't, but my computer is a bit far from a toaster. You're right: it might be a problem. So I rewrote the whole thing to be less computationally intensive. I've been testing it with a couple of dozens NPCs in the woods for quite a while. Now it seems to be working fine.